repository/user_repository.go

```diff
@@ -14,6 +15,8 @@ const (
 
 type UserRepository interface {
 	FindUser(ctx context.Context, email string) (*domain.User, error)
+	CreateUserQuery(ctx context.Context, name, email, password string, isAdmin bool) (*domain.User, error)
```

Comment:
Rename the method to "CreateUser"

and the input arguments should be: domain.User
because, we are using almost all the fields of that struct